app/models/template.rb

```diff
@@ -84,13 +84,16 @@ def ignore_default
 
   # Set attributes based on template +text and metadata found in this +text
   # the metadata parsing can be adjusted using +options
-  def import_without_save(text, options = {})
+  def import_without_save(name, text, options = {})
```

Comment:
You are changing a method signature ( adding parameter), what has big consequences on ALL the callers, not just one and I believe this method is called from mutiple places.
Given it is just for validation of the name, that is not great.

Reply:
You are right, that is why it is a draft. 
I'll find a different way :)